src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConversionTests.cs

```diff
@@ -1181,5 +1181,32 @@ .maxstack  4
   IL_002a:  ret
 }");
         }
+
+        [Fact, WorkItem(31587, "https://github.com/dotnet/roslyn/issues/31587")]
+        public void NullableDecimalToEnumConversion()
+        {
+            var source = @"
+enum E { }
+class C
+{
+    static E F(decimal? d) => (E)d;
+}
+";
+            CompileAndVerify(source).VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(31587, "https://github.com/dotnet/roslyn/issues/31587")]
+        public void MethodGroupConversion()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void F1() {}
+    static Action A => (Action)F1;
```

Comment:
This case is already likely covered quite well in other code. Given the problems around nullable I think we should focus on that type of method group conversion here. Example:

``` csharp
void M(decimal? d) {
  Func<string> f = d. ToString;
}
``` 

This works on my machine but I think it's the most reasonable test we can do here for the method group conversion case.

Reply:
Makes sense, will change the test.

---
In reply to: [247725681](https://github.com/dotnet/roslyn/pull/32372#discussion_r247725681) [](ancestors = 247725681)